src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttServerHandler.java

```diff
@@ -25,6 +25,7 @@
  */
 public class MqttServerHandler extends SimpleChannelInboundHandler<MqttMessage> {
     private static final Logger logger = LoggerFactory.getLogger(MqttServerHandler.class);
+    private MqttKafkaMapper mapper = new MqttKafkaMapper(new ArrayList<>());
```

Comment:
You are initializing the mapper with an empty list of rules, and I don't see them to be loaded later. How is this going to work?

Reply:
@ppatierno is there any reason we should use List instead of array list?

Reply:
> the mapper is not meant to be used withing the server yet.

So I would remove this changes to the server for now. Let's make only changes which are meaningful to the PR.

> is there any reason we should use List instead of array list?

In inheritance, you usually want always have the reference to be declared as an interface or base/abstract class. If you want to move to an ArrayList to LinkedList for example it's much simpler. Use a concrete class as declaration only if you don't have an interface or base class. In any case you can cast if needed.